app/Phragile/Burndown.php

```diff
@@ -0,0 +1,107 @@
+<?php
+namespace Phragile;
+
+class Burndown {
+	// TODO: ideally these should come from a cached call to maniphest.querystatuses
+	private static $STATUS_OPEN = ['stalled', 'open'];
+
+	public function __construct(\Sprint $sprint, TaskList $tasks, PhabricatorAPI $phabricator)
+	{
+		$this->sprint = $sprint;
+		$this->tasks = $tasks;
+		$this->phabricator = $phabricator;
+		$this->days = $this->daysInSprint();
+	}
+
+	private function daysInSprint()
+	{
+		$days = [];
+
+		for ($day = strtotime($this->sprint->sprint_start);
+		     $day <= strtotime($this->sprint->sprint_end);
+		     $day += 60*60*24)
+		{
+			$days[] = $day;
+		}
+
+		return $days;
+	}
+
+	public function getDays($format = 'M j')
+	{
+		return array_map(function($day) use($format)
+		{
+			return date($format, $day);
+		}, $this->days);
+	}
+
+	public function closedPerDay()
+	{
+		$closedPerDay = array_fill_keys(array_merge($this->getDays('Y-m-d'), ['before', 'after']), 0);
```

Comment:
They're not really different types of values. The resulting array describes the number of closed story points at a certain time. Tasks can be closed at a day of the sprint, `before` or `after`. I agree that it looks confusing and having all of it in one array wasn't as convenient as I thought anyway so I'll change that.